code/modules/power/supermatter/supermatter.dm

```diff
@@ -91,9 +91,12 @@
 	. = ..()
 
 /obj/machinery/power/supermatter_shard/proc/explode()
-	investigate_log("has exploded.", "supermatter")
-	explosion(get_turf(src), explosion_power, explosion_power * 2, explosion_power * 3, explosion_power * 4, 1, 1)
+	investigate_log("has collapsed into a singularity.", "supermatter")
+	var/turf/T = get_turf(src)
 	qdel(src)
+	if(T)
+		var/obj/singularity/S = new(T)
```

Comment:
I think this will have to wait until a later PR, unless @KorPhaeron wants to add something to this PR?